packages/web-app-external/src/App.vue

```diff
@@ -84,15 +87,27 @@ export default {
     appName() {
       return this.$route.params.appName
     },
-    fileId() {
-      return this.$route.params.fileId
+    filePath() {
+      return this.$route.params.filePath
     }
   },
+  mounted() {
+    if (this.$route.params.filePath && this.$route.params.appName)
+      document.title = `${this.$route.params.filePath.split('/').pop()} - ${
+        this.$route.params.appName
+      }`
+    else if (this.$route.params.appName) document.title = `${this.$route.params.appName}`
+    else if (this.$route.params.filePath) document.title = `${this.$route.params.filePath}`
+  },
   async created() {
     await unauthenticatedUserReady(this.$router, this.$store)
 
     this.loading = true
 
+    // get fileId by filePath
+    const resource = await this.$client.files.fileInfo(this.filePath, DavProperties.Default)
+    this.fileId = buildResource(resource).id
```

Comment:
In the short term: this should also go through the `useAppDefaults` composable and use getFileInfo there.

In the long term: If I understood @kulmann correctly, we will add the fileId to the query params anyway, so we could also add it to the `FileContext` and it would make this call completely obsolete.